### ISSUE TITLE
[Bug] 'invalid latitude' error when using `FillStyleExtension` with `MVTLayer` (8.8.21)

### ISSUE THREAD

**Issue by edtjones** [visgl/deck.gl#7566]
### Description

I'm trying to implement the `FillPatternExtension` for an `MTVLayer`, and I'm getting an `invalid latitude` error from the `projectFlat` method in `WebMercatorViewport` in Deck.gl 8.8.21.

This has definitely worked in earlier versions. I found issue #4718 from 2020 which explains the correct use of the extension with MVTLayer and forked the working Codepen by @Pessimistress in that thread (https://github.com/visgl/deck.gl/issues/4718#issuecomment-650356883)

I updated the dependencies to Deck.gl 8.8.21 and I get the same error as I'm experiencing in my application:

https://codepen.io/ed-humap/pen/RwBoBra?editors=1111

Any workarounds would be gratefully received.

### Flavors

- [X] React
- [ ] Python/Jupyter notebook
- [ ] MapboxLayer
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Expected Behavior

The FillStyleExtension should work with MVTLayer, following this merge: https://github.com/visgl/deck.gl/pull/4720

A working Codepen with Deck.gl 8.2.0-beta2 is here:

https://codepen.io/Pessimistress/pen/qBbrbaw

### Steps to Reproduce

This is the non-working version with Deck.gl 8.8.21:

https://codepen.io/ed-humap/pen/RwBoBra?editors=1111

### Environment

- Framework version: deck.gl@8.8.21
- Browser: All, but I'm using Firefox 108.0.1
- OS: MacOS 11.7

### Logs

```
"deck: drawing SolidPolygonLayer({id: 'MVTLayer-164-396-10-polygons-fill'}) to screen: invalid latitude"

"deck: drawing ScatterplotLayer({id: 'MVTLayer-164-396-10-points-circle'}) to screen: invalid latitude"

"deck: drawing SolidPolygonLayer({id: 'MVTLayer-165-396-10-polygons-fill'}) to screen: invalid latitude"

"deck: drawing ScatterplotLayer({id: 'MVTLayer-165-396-10-points-circle'}) to screen: invalid latitude"

"deck: drawing SolidPolygonLayer({id: 'MVTLayer-166-396-10-polygons-fill'}) to screen: invalid latitude"

"deck: drawing ScatterplotLayer({id: 'MVTLayer-166-396-10-points-circle'}) to screen: invalid latitude"
```

<img width="323" alt="Screenshot 2023-01-06 at 14 43 15" src="https://user-images.githubusercontent.com/4377783/211036876-e831072c-2711-4275-b81c-27fa6abe93e2.png">

**Comment by Pessimistress** [comment]
Thank you for reporting. Fixed in v8.8.22.

**Comment by edtjones** [comment]
You work so hard on this codebase @pessimistress - I’m very grateful 🙂 Thankyou for all your work! On 7 Jan 2023, at 19:50, Xiaoji Chen ***@***.***> wrote:﻿
Closed #7566 as completed.
